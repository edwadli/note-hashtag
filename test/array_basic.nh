x = { 1 2 3 }
PrintEndline (StringOfInt x.(0))
PrintEndline (StringOfInt x.(1))
PrintEndline (StringOfInt x.(2))

y = { 4.0 4.5 5.0 }
PrintEndline (StringOfFloat y.(0))
PrintEndline (StringOfFloat y.(1))
PrintEndline (StringOfFloat y.(2))

z = { "6 str" "7 str" "8 str" }
PrintEndline z.(0)
PrintEndline z.(1)
PrintEndline z.(2)

u = { () () () }
if u.(0) == () then PrintEndline "array of unit OK" else PrintEndline ":("

type ty = {
  a = "hi"
}
ta = { (init ty) (init ty) }
ti = ta.(0)
PrintEndline ti$a
