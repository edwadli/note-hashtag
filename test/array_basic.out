1
2
3
4.000000
4.500000
5.000000
6 str
7 str
8 str
array of unit OK
hi
